core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -343,23 +344,25 @@ public NativeImageBuildItem build(NativeConfig nativeConfig, NativeImageSourceJa
     public static List<String> setupContainerBuild(NativeConfig nativeConfig,
             Optional<ProcessInheritIODisabled> processInheritIODisabled, Path outputDir) {
         List<String> nativeImage;
-        String containerRuntime = nativeConfig.containerRuntime.orElse("docker");
+        final ContainerRuntime containerRuntime = nativeConfig.containerRuntime
+                .orElseGet(NativeImageBuildStep::detectContainerRuntime);
+        log.infof("Using %s to build the native image", containerRuntime.getExecutableName());
```

Comment:
Consider this wording instead?

`Using container engine %s to run the native image builder`, ....

Reply:
Thanks, I reworded a bit differently from your suggestion. I prefer to avoid using "container engine" since the existing parameter is named "container runtime"

Reply:
Fine with me.